.circleci/config.yml

```diff
@@ -294,244 +267,189 @@ commands:
-              | cut -c 1- \
-              | sed 's@src/test/java/@@' \
-              | sed 's@/@.@g' \
-              | sed 's/\.[^.]*$//' \
-              | circleci tests split --split-by=timings --timings-type=classname --index=$CIRCLE_NODE_INDEX)
-            GRADLE_ARGS=$(echo $CLASSNAMES | awk '{for (i=1; i<=NF; i++) print "--tests", $i }')
-            gradle :test $GRADLE_ARGS
-      - run:
-          # Lint last; it's more important to surface test failures early.
-          name: Java linting
-          working_directory: ~/workbench/api
-          command: |
-            if [ "$CIRCLE_NODE_INDEX" == 0 ]; then
-              ./gradlew spotlessCheck
-            fi
-      - store_test_results:
-           path: ~/workbench/api/build/test-results/test
-      # Save JUnit test results in artifacts because it provides the timing info on tests.
-      - store_artifacts:
-          path: ~/workbench/api/build/test-results/test
-          destination: JunitTestResult
-      - manage_api_cache:
-          save: true
+      - run-api-test:
+          additional_steps:
+            - run:
+                name: Validate Swagger definitions
+                working_directory: ~/workbench/api
+                command: |
+                  if [ "$CIRCLE_NODE_INDEX" == 0 ]; then
+                    ./project.rb validate-swagger --project-prop verboseTestLogging=yes
+                  fi
+            - run:
+                name: Run Java genomics unit tests
+                working_directory: ~/workbench/api
+                command: |
+                  CLASSNAMES=$(circleci tests glob "genomics/src/test/java/**/*Test.java" \
+                    | cut -c 1- \
+                    | sed 's@genomics/src/test/java/@@' \
+                    | sed 's@/@.@g' \
+                    | sed 's/\.[^.]*$//' \
+                    | circleci tests split --split-by=timings --index=$CIRCLE_NODE_INDEX)
+                  GRADLE_ARGS=$(echo $CLASSNAMES | awk '{for (i=1; i<=NF; i++) print "--tests", $i }')
+                  gradle -p genomics test $GRADLE_ARGS
+            - run:
+                name: Run Java api unit tests
+                working_directory: ~/workbench/api
+                # For script logics and explanation, see README.
+                command: |
+                  CLASSNAMES=$(circleci tests glob "src/test/java/**/*Test.java" "src/test/java/**/*Test.kt" \
+                    | cut -c 1- \
+                    | sed 's@src/test/java/@@' \
+                    | sed 's@/@.@g' \
+                    | sed 's/\.[^.]*$//' \
+                    | circleci tests split --split-by=timings --index=$CIRCLE_NODE_INDEX)
+                  GRADLE_ARGS=$(echo $CLASSNAMES | awk '{for (i=1; i<=NF; i++) print "--tests", $i }')
+                  gradle :test $GRADLE_ARGS
+            - run:
+                # Lint last; it's more important to surface test failures early.
+                name: Java linting
```

Comment:
Pleas name these steps all with consistent form (as they appear in a list). I think `Lint Java Code` would be good.